src/backend/distributed/utils/background_jobs.c

```diff
@@ -0,0 +1,1238 @@
+ */
+static void
+ConsumeTaskWorkerOutput(shm_mq_handle *responseq, BackgroundTask *task, bool *hadError,
+						bool *cancelled)
+{
+	/*
+	 * Message-parsing routines operate on a null-terminated StringInfo,
+	 * so we must construct one.
+	 */
+	StringInfoData msg = { 0 };
+	initStringInfo(&msg);
+
+	for (;;)
+	{
+		resetStringInfo(&msg);
+
+		/*
+		 * Get next message. Currently blocking, hen multiple backends get implemented it
+		 * should switch to a non-blocking receive
+		 */
+		Size nbytes = 0;
+		void *data = NULL;
+		const bool noWait = false;
+		shm_mq_result res = shm_mq_receive(responseq, &nbytes, &data, noWait);
+
+		if (res != SHM_MQ_SUCCESS)
+		{
+			break;
+		}
+
+		appendBinaryStringInfo(&msg, data, nbytes);
+
+		char msgtype = pq_getmsgbyte(&msg);
+		switch (msgtype)
+		{
+			case 'E': /* ERROR */
+			{
+				if (hadError)
+				{
+					*hadError = true;
+				}
+				__attribute__((fallthrough));
+			}
+
+			case 'N': /* NOTICE */
+			{
+				ErrorData edata = { 0 };
+				StringInfoData display_msg = { 0 };
+
+				pq_parse_errornotice(&msg, &edata);
+				initStringInfo(&display_msg);
+				bgw_generate_returned_message(&display_msg, edata);
+
+				if (edata.sqlerrcode == ERRCODE_QUERY_CANCELED)
+				{
+					if (cancelled)
+					{
+						*cancelled = true;
+					}
+				}
+
+				StringInfoData fullMessage = { 0 };
+				initStringInfo(&fullMessage);
+				if (task->message)
+				{
+					appendStringInfoString(&fullMessage, task->message);
+				}
+				appendStringInfoString(&fullMessage, display_msg.data);
+				appendStringInfoChar(&fullMessage, '\n');
+
+				/*
+				 * the task might live in a separate context, hence we find its context
+				 * and allocate a copy of the message in there
+				 */
+				MemoryContext taskContext = GetMemoryChunkContext(task);
+				task->message = MemoryContextStrdup(taskContext, fullMessage.data);
+
+				pfree(display_msg.data);
+				pfree(fullMessage.data);
+
+				break;
+			}
+
+			case 'T':
+			{
+				break;
+			}
+
+			case 'C': /* completed? */
+			{
+				const char *tag = pq_getmsgstring(&msg);
+
+				char *nonconst_tag = pstrdup(tag);
+
+				/* append the nonconst_tag to the task's message*/
+				StringInfoData fullMessage = { 0 };
+				initStringInfo(&fullMessage);
+				if (task->message)
+				{
+					appendStringInfoString(&fullMessage, task->message);
+				}
+				appendStringInfoString(&fullMessage, nonconst_tag);
+				appendStringInfoChar(&fullMessage, '\n');
+
+				/*
+				 * the task might live in a separate context, hence we find its context
+				 * and allocate a copy of the message in there
+				 */
+				MemoryContext taskContext = GetMemoryChunkContext(task);
+				task->message = MemoryContextStrdup(taskContext, fullMessage.data);
+				pfree(fullMessage.data);
+
+				pfree(nonconst_tag);
+
+				task->status = BACKGROUND_TASK_STATUS_DONE;
+				break;
+			}
+
+			case 'A':
+			case 'D':
+			case 'G':
+			case 'H':
+			case 'W':
+			case 'Z':
```

Comment:
Maybe add comments about what these mean or at least a link to the docs where you can find out.

Reply:
Once you know where they come from let me know ;)
https://github.com/citusdata/pg_cron/blob/6bf8443fb39226fec787eeff656ee5d147573116/src/pg_cron.c#L1951-L1957

Reply:
I think it's the letters that define the message types from here: https://www.postgresql.org/docs/current/protocol-message-formats.html